RecoEgamma/EgammaTools/python/egammaObjectModificationsInMiniAOD_cff.py

```diff
@@ -170,3 +183,6 @@ def appendEgammaHIPhotonIsolationModifier(modifiers):
 
 from Configuration.ProcessModifiers.pp_on_AA_cff import pp_on_AA
 pp_on_AA.toModify(egamma_modifications, appendEgammaHIPhotonIsolationModifier)
+
+from Configuration.ProcessModifiers.photonDRN_cff import photonDRN
+photonDRN.toModify(egamma_modifications, appendPhotonDRNModifier)
```

Comment:
what happens if photonDRN is enabled but enableSonicTriton is not enabled? Is it defined to be an error state, so photonDRN must imply also enableSonicTriton?

Reply:
Indeed photonDRN without enableSonicTriton will error out before processing the first event. Is there a way to mark this interdependence explicitly in the code?

Reply:
I'm actually not sure if there's a way to enforce this. @makortel ?

Reply:
I'm not sure if I understood the question correctly, but one can do
```suggestion
(enableSonicTriton & photonDRN).toModify(egamma_modifications, appendPhotonDRNModifier)
```
in which case the modification is applied only if both `enableSonicTriton` and `photonDRN` modifiers are enabled.

Reply:
Yes, I think that's the practical approach right now. The longer-term question is whether this can be enforced more automatically, e.g. disallow use of the photonDRN modifier if enableSonicTriton isn't also used (rather than having to put `(enableSonicTriton & photonDRN)` everywhere manually).

Reply:
Might it not be preferable to just have an error for incorrect configuration? I feel like the desired behavior when photonDRN is enabled without enableSonicTriton would be an error rather than silently just refusing to turn on the DRN.

Reply:
Obviously in an ideal world it would be a custom error, but even as currently set up it would crash before the first event with a message along the lines of "could not find the Sonic Triton service"

Reply:
How about creating a `ModifierChain` that includes `enableSonicTriton` and `photonDRN` modifier, and pass the `ModifierChain` to the `--procModifiers`?

Reply:
A `ModifierChain` is useful for defining a workflow, but doesn't stop someone from specifying `photonDRN` by itself as the argument to `--procModifers`. Of course, as @ssrothman points out, when exactly a workflow experiences an error may not matter too much in practice.

Reply:
Maybe we could define a modifier named something like `_photonDRN` and then a public-facing modifier `photonDRN` which is a `ModifierChain` with both `_photonDRN` and `enableSonicTriton`?